twilio/http/http_client.py

```diff
@@ -78,6 +78,11 @@ def request(
         elif timeout <= 0:
             raise ValueError(timeout)
 
+        if headers:
+            if "Requires-Authentication" in headers:
+                headers.pop("Requires-Authentication", None)
+                auth = None
```

Comment:
Test cases

1. Initialize basic call api with basic api then call no auth api - working fine
2. Then call basic again - test this

Reply:
This case is working fine